core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -226,6 +227,33 @@
   public static final long SERVER_KEEPALIVE_TIME_NANOS_DISABLED = Long.MAX_VALUE;
 
   /**
+   * The default proxy detector.
+   */
+  public static final ProxyDetector DEFAULT_PROXY_DETECTOR = new ProxyDetectorImpl();
+
+  /**
+   * A proxy detector that always claims no proxy is needed.
+   */
+  public static final ProxyDetector NOOP_PROXY_DETECTOR = new ProxyDetector() {
```

Comment:
optional:

this only appears to be used in tests.  You might consider pulling this out into its own static class, putting the instance inside of it, and referencing it from tests like that.  

Then the regular code won't have to initialize this anonymous class in prod code.

Reply:
This is no longer true because GAE+jdk7 will use this noop detector.